Random failing IProjectTest.testProjectDeletionOpenDefaultOutOfSync / IProjectTest.testProjectDeletionOpenUserDefinedOutOfSync

The tests `org.eclipse.core.tests.resources.IProjectTest.testProjectDeletionOpenDefaultOutOfSync`  and `org.eclipse.core.tests.resources.IProjectTest.testProjectDeletionOpenDefaultOutOfSync` are randomly failing in recent builds. Both fails are related to validating a deletion with out-of-sync artifacts, which is why I put them into the same issue. They occur on different OS / file systems (Windows / MacOS).

https://github.com/eclipse-platform/eclipse.platform/actions/runs/4765749986/jobs/8477798130:
```
junit.framework.AssertionFailedError: 6.1
	at junit.framework.Assert.fail(Assert.java:57)
	at junit.framework.Assert.assertTrue(Assert.java:22)
	at junit.framework.Assert.assertFalse(Assert.java:39)
	at junit.framework.TestCase.assertFalse(TestCase.java:208)
	at org.eclipse.core.tests.resources.IProjectTest.testProjectDeletionOpenDefaultOutOfSync(IProjectTest.java:1630)
```

https://github.com/eclipse-platform/eclipse.platform/actions/runs/4785187170/jobs/8507931930
```
java.lang.AssertionError: expected org.eclipse.core.runtime.CoreException to be thrown, but nothing was thrown
	at org.junit.Assert.assertThrows(Assert.java:1028)
	at org.junit.Assert.assertThrows(Assert.java:981)
	at org.eclipse.core.tests.resources.IProjectTest.testProjectDeletionOpenUserDefinedOutOfSync(IProjectTest.java:1932)
```

--- a/resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/IProjectTest.java
+++ b/resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/IProjectTest.java
@@ -1372,6 +1372,7 @@ public class IProjectTest extends ResourceTest {
 		description.setLocationURI(projectStore.toURI());
 		ensureExistsInWorkspace(project, description);
 		ensureExistsInWorkspace(file, true);
+		waitForRefresh();
 		ensureExistsInFileSystem(otherFile);
 		fileStore = ((Resource) file).getStore();
 		otherFileStore = ((Resource) otherFile).getStore();
@@ -1623,6 +1624,7 @@ public class IProjectTest extends ResourceTest {
 		 * Delete content = DEFAULT
 		 * =======================================================================*/
 		ensureExistsInWorkspace(project, true);
+		waitForRefresh();
 		ensureExistsInFileSystem(file);
 		projectStore = ((Resource) project).getStore();
 		fileStore = ((Resource) file).getStore();
@@ -1923,6 +1925,7 @@ public class IProjectTest extends ResourceTest {
 		description.setLocationURI(projectStore.toURI());
 		ensureExistsInWorkspace(project, description);
 		ensureExistsInWorkspace(file, true);
+		waitForRefresh();
 		ensureExistsInFileSystem(otherFile);
 		fileStore = ((Resource) file).getStore();
 		otherFileStore = ((Resource) otherFile).getStore();

